cdap-cli/src/main/java/co/cask/cdap/cli/command/GetStreamStatsCommand.java

```diff
@@ -0,0 +1,240 @@
+    }
+
+    // get a list of stream events and calculates various statistics about the events
+    ListenableFuture<ExploreExecutionResult> resultsFuture = queryClient.execute(
+      "SELECT * FROM cdap_stream_" + streamId + " LIMIT " + limit);
+    ExploreExecutionResult results = resultsFuture.get(1, TimeUnit.MINUTES);
+    List<ColumnDesc> schema = results.getResultSchema();
+
+    // apply StatsProcessors to every element in every row
+    while (results.hasNext()) {
+      QueryResult row = results.next();
+      for (int i = 0; i < row.getColumns().size(); i++) {
+        Object column = row.getColumns().get(i);
+        ColumnDesc columnDesc = schema.get(i);
+        String columnName = columnDesc.getName();
+        Set<StatsProcessor> processors = processorMap.get(columnName);
+        if (processors != null) {
+          for (StatsProcessor processor : processors) {
+            processor.process(column);
+          }
+        }
+      }
+    }
+
+    // print report
+    for (ColumnDesc columnDesc : schema) {
+      output.printf("column: %s, type: %s", columnDesc.getName(), columnDesc.getType());
+      output.println();
+      Set<StatsProcessor> processors = processorMap.get(columnDesc.getName());
+      if (processors != null && !processors.isEmpty()) {
+        for (StatsProcessor processor : processors) {
+          processor.printReport(output);
+        }
+        output.println();
+      } else {
+        output.println("No statistics available");
+        output.println();
+      }
+    }
+
+  }
+
+  private String cdapSchemaColumName2HiveColumnName(String streamId, String schemaColumName) {
+    return "cdap_stream_" + streamId + "." + schemaColumName;
+  }
+
+  private Set<StatsProcessor> getProcessorsForType(Schema.Type type, List<Schema> unionSchemas) {
+    ImmutableSet.Builder<StatsProcessor> result = ImmutableSet.builder();
+
+    boolean isInt = isTypeOrInUnion(Schema.Type.INT, type, unionSchemas);
+    boolean isLong = isTypeOrInUnion(Schema.Type.LONG, type, unionSchemas);
+    boolean isString = isTypeOrInUnion(Schema.Type.STRING, type, unionSchemas);
+
+    if (isInt || isLong || isString) {
+      result.add(new CountUniqueProcessor());
+    }
+
+    if (isInt || isLong) {
+      result.add(new HistogramProcessor());
+    }
+
+    return result.build();
+  }
+
+  private boolean isTypeOrInUnion(Schema.Type desiredType, Schema.Type type, List<Schema> unionSchemas) {
+    if (desiredType.equals(type)) {
+      return true;
+    }
+
+    for (Schema unionSchema : unionSchemas) {
+      if (desiredType == unionSchema.getType()) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  @Override
+  public String getPattern() {
+    return String.format("get stream-stats <%s> [<%s>]", ArgumentName.STREAM, ArgumentName.LIMIT);
+  }
+
+  @Override
+  public String getDescription() {
+    return "Gets statistics for a " + ElementType.STREAM.getPrettyName() + ". " +
```

Comment:
Another place we could use a function to return the correct article ("a" or "an")